src/toolkit.cpp

```diff
@@ -224,53 +225,13 @@ FileFormat Toolkit::IdentifyInputFrom(const std::string &data)
         // <score-timewise> == root node for time-wise organization of MusicXML data
         // <opus> == root node for multi-movement/work organization of MusicXML data
 
-        if (initial.find("<mei ") != std::string::npos) {
+        if (std::regex_search(initial, std::regex("<(mei|music|pages)[\\s\\n>]"))) {
```

Comment:
Using a raw string means you don't have to double-escape the `\s` and `\n`:

`R"(<(mei|music|pages)[\s\n>])"`

Reply:
I used to use raw strings in verovio code, but the clang-format program mangled them: it placed a space between the `R` and `"`.  So every time I ran clang-format, I had to go back and remove the space, since I could not find any option to suppress that behavior (and clang-format does not or did not understand raw string).

Reply:
Oh, good to know!

Reply:
Looks like you might be able to control this behaviour setting the LanguageStandard behaviour to C++11:

http://clang-developers.42468.n3.nabble.com/clang-format-raw-string-literals-td4042533.html

`LanguageStandard: c++11`

Reply:
I think you can do
```cpp
// clang-format off
do stuff
// clang-format on

Reply:
```
// clang-format off
#include "vrv.h"
...

} // namespace vrv
// clang-format on
```

:-D

Reply:
Another consideration for the regular expression is that `\n` is one of the characters represented by `\s`, so the `\n` is redundant, and the regular expression can be simplified to:

```
"<(mei|music|pages)[\\s>]"
```

Reply:
:-P.  See the new issue I am making in case my note below on this page gets lost.